src/Pirouette/Term/ToTla.hs

```diff
@@ -844,8 +898,9 @@ tlaAppNamePref pref n args = do
   arity <- case mty of
              Just ty -> return (arity $ tlaTyDropAll ty)
              Nothing -> tlaPure (logWarn $ "Undeclared type for " ++ pref ++ show (toString n)) >> return 0
-  let (tas, das) = L.splitAt arity args
-  return $ tlaFunApp (tlaOpApp (tlaIdentPrefixed pref n) tas) das
+  if arity >= length args
+  then pure $ tlaOpApp (tlaIdentPrefixed pref n) args
+  else pure $ tlaOpApp (applyFunIdent $ length args - arity) (tlaIdentPrefixed pref n : args)
```

Comment:
Another question; 

It seems like you're dropping the application of arguments to the operator `n`. I'll try to elaborate.
Before defunctionalization, say `n` had arity 3 but it have been given 5 arguments. 
This means that `n` must have been declared as a TLA+ operator that takes 3 arguments, hence
```haskell
return $ tlaFunApp (tlaOpApp (tlaIdentPrefixed pref n) tas) das
```
would return: `prefN(x1,x2,x3)[x4, x5]`

The first three arguments are given to the TLA+ operator, and whatever left must be applied to whatever the operator returns.

Reply:
I am not sure to understand what you are saying.
Are you stating that it would be better to write `apply2(f(x1,x2,x3),x4,x5)` rather than `apply5(f,x1,x2,x3,x4,x5)`? If so, I would be glad to hear your arguments, because I have the same feeling, but no justifications.
Or are you simply pointing that `apply5(f,x1,x2,x3,x4,x5)` is "ill-typed" in the sense that the first argument of `apply5` should not be an operator? The solution to keep this approach would be to write `apply5(curry3(f),x1,x2,x3,x4,x5)`. If someone have arguments regarding this translation, I would be glad to hear it too.

Reply:
I was thinking about it while nix is updating the version of Plutus, and I think that I have a strong argument against `apply5(curry3(f),x1,x2,x3,x4,x5)`.
The definition of `curry3` would have to be `curry3(g(_,_,_)) == [x1 \in a, x2 \in b, x3\in c |-> g(x1,x2,x3)]` which would require to carry the type information somewhere and would not avoid us the rabbit hole of uselessly constructing sets with millions of elements before a simple computation.

Reply:
Right, this is something I've missed, good catch. The fix seems to be easy, but do we have any contract code that'd exercise this? It's nice to have a test for this both for verifying the implementation makes sense and for future test suite.